File: src/test/java/com/tartner/domain/password/PasswordProjectorTest.java
package com.tartner.domain.password;

import com.tartner.IntegrationTestCategory;
import com.tartner.dancehours.querymodel.jpa.AggregatePasswordsEntity;
import org.junit.Before;
import org.junit.Test;
import org.junit.experimental.categories.Category;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
import org.springframework.test.context.transaction.TransactionConfiguration;
import org.springframework.transaction.annotation.Transactional;

import java.security.NoSuchAlgorithmException;
import java.util.UUID;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;
import static org.hamcrest.Matchers.notNullValue;

@RunWith(SpringJUnit4ClassRunner.class)
@ContextConfiguration(classes=StandardIntegrationTestConfiguration.class)
@ComponentScan(basePackages = { "com.tartner.domain" })
@Transactional
@TransactionConfiguration(defaultRollback=true)
@Category( IntegrationTestCategory.class )
public class PasswordProjectorTest {
    @Autowired private AggregatePasswordRepository repository;
    private PasswordProjector projector;

    @Before
    public void setUp() throws NoSuchAlgorithmException {
    }

    @Test
    public void checkEvent() throws Exception {
        projector = new PasswordProjector( repository );

        TestPasswordHolder holder = TestPasswordHolder.CreateDefaultTest();

        final UUID uuid = UUID.randomUUID();

        final PasswordSetEvent event = new PasswordSetEvent( uuid,
            holder.getPasswordHash(), holder.getSalt() );
        projector.handle( event );

        final AggregatePasswordsEntity passwordsEntity = repository.findOne( uuid );

        assertThat( passwordsEntity, notNullValue() );

        assertThat( passwordsEntity.getPasswordHash(), is( holder.passwordHash ) );
        assertThat( passwordsEntity.getSalt(), is( holder.salt ) );
    }

}

File: src/test/java/com/tartner/domain/password/PasswordProjectorTest.kt
package com.tartner.domain.password

import com.tartner.IntegrationTestCategory
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.`is`
import org.hamcrest.Matchers.notNullValue
import org.junit.Test
import org.junit.experimental.categories.Category
import org.junit.runner.RunWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.ComponentScan
import org.springframework.test.context.ContextConfiguration
import org.springframework.test.context.junit4.SpringJUnit4ClassRunner
import org.springframework.test.context.transaction.TransactionConfiguration
import org.springframework.transaction.annotation.Transactional
import java.util.*

@RunWith(SpringJUnit4ClassRunner::class)
@ContextConfiguration(classes = arrayOf(StandardIntegrationTestConfiguration::class))
@ComponentScan(basePackages = arrayOf("com.tartner.domain"))
@Transactional
@TransactionConfiguration(defaultRollback = true)
@Category(IntegrationTestCategory::class)
public open class PasswordProjectorTest {
    @Autowired private var repository: AggregatePasswordRepository? = null

    @Test @throws(Exception::class)
    public fun checkEvent() {
        if( repository == null ) throw IllegalStateException("repository not initialized")

        val projector = PasswordProjector(repository!!)

        val holder = TestPasswordHolder.CreateDefaultTest()

        val uuid = UUID.randomUUID()

        val event = PasswordSetEvent(uuid, holder.passwordHash, holder.salt)
        projector.handle(event)

        val passwordsEntity = repository!!.findOne(uuid)

        assertThat(passwordsEntity, notNullValue())

        assertThat<ByteArray>(passwordsEntity.passwordHash, `is`(holder.passwordHash))
        assertThat<ByteArray>(passwordsEntity.salt, `is`(holder.salt))
    }
}
